modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -148,7 +159,17 @@ public void onWrite(Iterator<CommandClosure<WriteCommand>> iterator) {
         iterator.forEachRemaining((CommandClosure<? extends WriteCommand> clo) -> {
             Command command = clo.command();
 
-            // LOG.info("CMD {}", command.getClass().getName());
+            if (command instanceof SafeTimePropagatingCommand) {
+                SafeTimePropagatingCommand cmd = (SafeTimePropagatingCommand) command;
+                long proposedSafeTime = cmd.safeTime().longValue();
+
+                if (proposedSafeTime > maxObservableSafeTimeVerifier) {
+                    maxObservableSafeTimeVerifier = proposedSafeTime;
+                } else {
+                    assert false : "Safe time reordering detected [current=" + maxObservableSafeTimeVerifier
+                            + ", proposed=" + proposedSafeTime + "]";
+                }
```

Comment:
> it would be better to remove it altogether.

I don't think so. Given assertion will ease tests failures analysis. In case of unexpected data we will see whether there was a reordering or not.

>  assert roposedSafeTime <= maxObservableSafeTimeVerifier : "Safe time reordering detected [current=" + maxObservableSafeTimeVerifier + ", proposed=" + proposedSafeTime + "].";

maxObservableSafeTimeVerifier = proposedSafeTime;

Agreed. Fixed. Except the part that proposed should be greater than maxObservableSafeTimeVerifier